app/helpers/dams_objects_helper.rb

```diff
@@ -688,6 +688,43 @@ def grabRestrictedText(data)
 
   end
 
+  #---
+  # Check to see if an object has a "metadataDisplay or localDisplay otherRights"
+  #
+  # @return An HTML string if an object has a "metadataDisplay or localDisplay otherRights", nil otherwise
+  #---
+
+  def grab_access_text(document)
+    result = nil
+    access_group = document['read_access_group_ssim'] # "public" > "local" > "dams-curator" == "dams-rci" == default
+    data = document['otherRights_tesim']
+    unless data.nil? && access_group.nil?
+      if access_group.include?('local') && !data.nil?
+        data.each do |datum|
+          if datum.include?('localDisplay') || datum.include?('metadataDisplay')
+            result = "<h3>Restricted View</h3><p>#{get_attribution_note(document['otherNote_json_tesim'])}</p>".html_safe
+          end
+        end
+      end
+    end
+    result
+  end
+
+  def get_attribution_note(data)
+    result = 'Content not available. Access may granted for research purposes at the discretion of the UC San Diego Library. For more information please contact the '
+    program_email = { 'Digital Library Development Program' => 'dlp@ucsd.edu', 'Special Collections & Archives' => 'spcoll@ucsd.edu', 'Research Data Curation Program' => 'research-data-curation@ucsd.edu'}
+    unless data.nil?
```

Comment:
you could probably do a `return result unless data` here and drop the conditional